app/controllers/YourTaxCodeController.scala

```diff
@@ -52,23 +56,47 @@ trait YourTaxCodeController extends TaiBaseController
               TaiSuccessResponseWithPayload(taxCodeIncomes: Seq[TaxCodeIncome]) <- taxAccountService.taxCodeIncomes(nino, year)
               scottishTaxRateBands <- taxAccountService.scottishBandRates(nino, year, taxCodeIncomes.map(_.taxCode))
             } yield {
-              val taxCodeViewModel = TaxCodeViewModel(taxCodeIncomes, scottishTaxRateBands, year)
+              val taxCodeViewModel = TaxCodeViewModel.applyTaxCodeIncomes(taxCodeIncomes, scottishTaxRateBands, year)
               Ok(views.html.taxCodeDetails(taxCodeViewModel))
             }
           }
   }
 
-  def prevTaxCodes(year: TaxYear): Action[AnyContent] = taxCodes(year)
+  def prevTaxCodes(year: TaxYear): Action[AnyContent] = authorisedForTai(personService).async {
```

Comment:
Not sure if this comment is relevant still. We should probably be passing the responsibility of what to do in each year down to the service. However because current year and previous years get the data from different places I don't know if there is much to do to remove the duplication